mapic/mistapiconnector_app.go

```diff
@@ -506,18 +525,66 @@ func (mc *mac) reconcileLoop(ctx context.Context) {
 		case <-ctx.Done():
 			return
 		case <-ticker.C:
-		case <-mc.multistreamUpdated:
+		case <-mc.streamUpdated:
 		}
 		mistState, err := mc.mist.GetState()
 		if err != nil {
 			glog.Errorf("error executing query on Mist, cannot reconcile err=%v", err)
 			continue
 		}
+		mc.reconcileStreams(mistState)
 		mc.reconcileMultistream(mistState)
 		mc.processStats(mistState)
 	}
 }
 
+func (mc *mac) reconcileStreams(mistState clients.MistState) {
+	for streamName, _ := range mistState.ActiveStreams {
+		// read map directly to avoid creating streamInfo object even for playback-only (pull) streams
+		mc.mu.RLock()
+		si := mc.streamInfo[mistStreamName2playbackID(streamName)]
+		mc.mu.RUnlock()
+
+		if isIngestStream(streamName, si, mistState) {
+			if si == nil {
+				var err error
+				si, err = mc.refreshStreamInfo(streamName)
+				if err != nil {
+					glog.Errorf("error refreshing stream info streamName=%s err=%v", streamName, err)
+					continue
+				}
+			}
+
+			mc.reconcileSingleStream(si)
+		}
```

Comment:
```suggestion
			if mistState.IsIngestStream(streamName) {
			si, err := mc.getStreamInfo(mistStreamName2playbackID(streamName))
			if err != nil {
				glog.Errorf("error getting stream info streamName=%s err=%v", streamName, err)
				continue
			}
			mc.reconcileSingleStream(si)
		}
```

How about this logic? Should work the same and is simpler.

Reply:
According to this comment, it is not the same: https://github.com/livepeer/dms-api/blob/835bb9ed3a912bd7ef9d3007b7ef9dc681101abd/mapic/mistapiconnector_app.go#L839

Do you think that case is not that important? I can change to that logic it's much cleaner 😌

Reply:
OK I've decided to go forward with it. If the docs I linked are correct, worst case we will wait 30 seconds until the next reconcile loop and it should proceed with the nuke there. We will not be nuking streams that have just started to the point of a "race condition" anywya.

Reply:
I think it's correct as I suggested (and as you changed it), because we iterate over Active Streams from Mist (so the scenario mentioned in the comment ☝️ is not important here).